client/state-hooks.ts

```diff
@@ -68,3 +68,28 @@ export function useForceUpdate(): () => void {
     forceUpdater({})
   }, [])
 }
+
+/**
+ * A hook which multiplexes the given ref with a local one. Mostly useful when needing to create a
+ * local ref in a component that already forwards a ref to another component below it.
+ */
+export function useMultiRef<T>(
+  ref: React.ForwardedRef<T>,
+): [elemRef: React.MutableRefObject<T | undefined>, setElemRef: (elem: T | null) => void] {
+  const elemRef = useRef<T>()
+  const setElemRef = useCallback(
+    (elem: T | null) => {
+      elemRef.current = elem ?? undefined
+      if (ref) {
```

Comment:
If we're dealing with `ForwardedRef` isn't this always true? (If you want to accept non-forwarded refs as well, I'd probably make the function param optional + use a different type)

Reply:
Typescript says that it might be `null`. And while I do think there's value in expanding this hook to include non-forwarded refs, I'll leave that as an exercise to a future reader if/when it's needed since I can't be arsed to fiddle with these types anymore :).

To be more serious, I'll do it if you think it should be done now.

Reply:
Well, in the future the easier answer to my question was to look at the types to ForwardedRef, which I didn't do before because I didn't have an editor open :P

```ts
type ForwardedRef<T> = ((instance: T | null) => void) | MutableRefObject<T | null> | null;
```

So yeah, it can (apparently) be null.

I don't think any further modifications are necessary to take non-forwarded refs, given that typing (although the named type does feel a bit strange)